### ISSUE TITLE
Consider adding Entity<T> yaml serialization support

### ISSUE THREAD

**Issue by ElectroJr** [space-wizards/RobustToolbox#5580]
Would be nice to have, but not trivial to support, because entity/component data is read sequentially. So you might be parsing an entity & its data-fields before the entity being referenced has been loaded, so you don't know yet know whether or not it has the component.

See #4855 and space-wizards/space-station-14/issues/24078

**Comment by PJB3005** [comment]
Yeah I'm not sure this is actually a good idea, kinda violates ECS IMO.
